File: December03/spiral1.py
n = int(input())
coord = (0, 0)
i = 1
x = 0
y = 0
row = 1

while i < n:
    while x < row:
        x += 1
        i += 1
        if i == n:
            coord = (x, y)
    while y < row:
        y += 1
        i += 1
        if i == n:
            coord = (x, y)
    while x > -row:
        x -= 1
        i += 1
        if i == n:
            coord = (x, y)
    while y > -row:
        y -= 1
        i += 1
        if i == n:
            coord = (x, y)
    row += 1

print(abs(coord[0]) + abs(coord[1]))

File: December03/spiral1.cpp
#include <iostream>

int main() {
    int n;
    std::cin >> n;
    int i = 1;
    int x = 0;
    int y = 0;
    int row = 1;
    bool solved = false;

    while (i < n) {
        while (x < row && !solved) {
            x++;
            solved = ++i == n;
        }
        while (y < row && !solved) {
            y++;
            solved = ++i == n;
        }
        while (x > -row && !solved) {
            x--;
            solved = ++i == n;
        }
        while (y > -row && !solved) {
            y--;
            solved = ++i == n;
        }
        row++;
    }
    std::cout << (abs(x) + abs(y)) << std::endl;
}